mzLib/MassSpectrometry/PeakIndexing/IndexingEngine.cs

```diff
@@ -194,11 +194,11 @@ public List<IIndexedPeak> GetXic(double m, int zeroBasedStartIndex, Tolerance pp
         /// <summary>
         /// A generic method performing peak tracing for all the peaks in an indexingEngine and trying to find all XICs.
         /// <returns> A list of ExtractedIonChromatogram objects representing all XICs that can be found in an indexingEngine </returns>
-        public virtual List<ExtractedIonChromatogram> GetAllXics(Tolerance peakFindingTolerance, int maxMissedScanAllowed, double maxRTRange, int numPeakThreshold)
+        public virtual List<ExtractedIonChromatogram> GetAllXics(Tolerance peakFindingTolerance, int maxMissedScanAllowed, double maxRTRange, int numPeakThreshold, double cutPeakDiscriminationFactor = -1)
```

Comment:
Make the cutPeakDiscriminationFactor a nullable double, and add a summary comment explaining what it does